lte/cloud/go/services/lte/servicers/indexer_servicer.go

```diff
@@ -0,0 +1,114 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package servicers
+
+import (
+	"context"
+	"fmt"
+
+	"magma/lte/cloud/go/lte"
+	"magma/lte/cloud/go/serdes"
+	lte_api "magma/lte/cloud/go/services/lte"
+	lte_models "magma/lte/cloud/go/services/lte/obsidian/models"
+	"magma/orc8r/cloud/go/serde"
+	"magma/orc8r/cloud/go/services/state/indexer"
+	"magma/orc8r/cloud/go/services/state/protos"
+	state_types "magma/orc8r/cloud/go/services/state/types"
+
+	"github.com/pkg/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	indexerVersion indexer.Version = 1
+)
+
+var (
+	indexerTypes = []string{lte.EnodebStateType}
+)
+
+type indexerServicer struct{}
+
+// NewIndexerServicer returns the state indexer for the lte service.
+//
+// Enodeb state is reported as ENB SN -> EnodebState json model for a given
+// networkID. Since multiple gateways can report this state, index the state to
+// add gatewayID as an additional primary key. This will allow differentiation
+// between state reported from different gateways for the same ENB SN.
+func NewIndexerServicer() protos.IndexerServer {
+	return &indexerServicer{}
+}
+
+func (i *indexerServicer) GetIndexerInfo(ctx context.Context, req *protos.GetIndexerInfoRequest) (*protos.GetIndexerInfoResponse, error) {
+	res := &protos.GetIndexerInfoResponse{
+		Version:    uint32(indexerVersion),
+		StateTypes: indexerTypes,
+	}
+	return res, nil
+}
+
+func (i *indexerServicer) Index(ctx context.Context, req *protos.IndexRequest) (*protos.IndexResponse, error) {
+	states, err := state_types.MakeStatesByID(req.States, serdes.State)
+	if err != nil {
+		return nil, err
+	}
+	stErrs, err := indexImpl(req.NetworkId, states)
+	res := &protos.IndexResponse{StateErrors: state_types.MakeProtoStateErrors(stErrs)}
+	return res, nil
+}
+
+func (i *indexerServicer) PrepareReindex(ctx context.Context, req *protos.PrepareReindexRequest) (*protos.PrepareReindexResponse, error) {
+	return &protos.PrepareReindexResponse{}, nil
+}
+
+func (i *indexerServicer) CompleteReindex(ctx context.Context, req *protos.CompleteReindexRequest) (*protos.CompleteReindexResponse, error) {
+	if req.FromVersion == 0 && req.ToVersion == 1 {
+		return &protos.CompleteReindexResponse{}, nil
+	}
+	return nil, status.Errorf(codes.InvalidArgument, "unsupported from/to for CompleteReindex: %v to %v", req.FromVersion, req.ToVersion)
+}
+
+func indexImpl(networkID string, states state_types.StatesByID) (state_types.StateErrors, error) {
+	return setEnodebState(networkID, states)
+}
+
+// setEnodebState stores EnodebState with reporterID as an additional PK
+func setEnodebState(networkID string, states state_types.StatesByID) (state_types.StateErrors, error) {
+	stateErrors := state_types.StateErrors{}
+	for id, st := range states {
+		// sanity check that we're operating on the correct state
```

Comment:
This check's superfluous -- in any case, if there was a bug in the state service's indexer code, the subsequent type conversion failure (below) would be sufficient notification that something was amiss